File: io-hotmoka-network-thin-client/io-hotmoka-network-thin-client-kt/src/main/kotlin/io/hotmoka/network/thin/client/models/responses/TransactionRestResponseModel.kt
package io.hotmoka.network.thin.client.models.responses

/**
 * Class which wraps a type response model
 * @param <T> the type response model
 */
class TransactionRestResponseModel<T>(
        /**
         * The runtime type of the response model
         */
        val type: String,
        /**
         * The response model which should be an instance of TransactionResponseModel.
         */
        val transactionResponseModel: T,
)

File: io-hotmoka-network-thin-client/io-hotmoka-network-thin-client-js/src/models/responses/TransactionRestResponseModel.ts
/**
 * Class which wraps a type response model
 *
 * @param <T> the type response model
 */
export class TransactionRestResponseModel<T> {

    /**
     * The response model which should be an instance of {@link TransactionResponseModel}.
     */
    transactionResponseModel: T

    /**
     * The runtime type of the response model
     */
    type: string

    constructor(transactionResponseModel: T, type: string) {
        this.transactionResponseModel = transactionResponseModel
        this.type = type
    }
}